File: back/src/main/java/com/openclassrooms/mddapi/models/TopicEntity.java
package com.openclassrooms.mddapi.models;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.HashSet;
import java.util.Set;

@Entity
@Table(name = "topics")
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class TopicEntity {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    @Column(name = "title")
    private String title;
    @Column(name = "description")
    private String description;
    @ManyToMany(mappedBy = "subscriptions", fetch = FetchType.LAZY)
    private Set<UserEntity> users = new HashSet<>();
}

File: front-app/src/app/domain/topics/Topic.entity.ts
import { PostEntity } from "../posts/Post.entity";

export class TopicEntity {
    id?: number;
    title: string;
    description: string;
    posts: PostEntity[];

    constructor(title: string, description: string, posts: PostEntity[]) {
        this.title = title;
        this.description = description;
        this.posts = posts;
    }
}